src/tasks/platforms/common-platform-tasks.ts

```diff
@@ -72,4 +79,54 @@ export namespace CommonPlatformTasks {
     })
   }
 
+  export function oAuthProvidersExists(flags: any): Listr.ListrTask {
+    let kube = new KubeHelper(flags)
+    return {
+      title: 'Verify Openshift oauth.',
+      enabled: () => isOpenshiftPlatformFamily(flags.platform) && isOAuthEnabled(flags),
+      task: async (ctx: any, task: any) => {
+        if (await kube.isOpenShift4()) {
+          const providers = await kube.getOpenshiftAuthProviders()
+          if (!providers || providers.length === 0) {
+            ctx.highlightedMessages.push(`❗ ${ansi.yellow('[WARNING]')} 'os-oauth' flag was disabled, because Openshift oauth hasn't got any identity providers. ${DOCS_LINK_HOW_TO_ADD_IDENTITY_PROVIDER_OS4}`)
+            flags['os-oauth'] = false
```

Comment:
Basically here
https://github.com/che-incubator/chectl/blob/master/src/api/kube.ts#L1133
we ignore flags if provided cr is used ^(

Reply:
Good catch.